From: SeongJae Park <sj@kernel.org>
Date: Wed, 15 Oct 2025 17:28:58 -0700
Subject: [PATCH] mm/damon/sysfs: put obsolete file input when commit

Note that the sysfs files are not automatically updated to follow the
core layer internal data structure for the obsolete targets, which are
removed from the internal data structure.  Updating the sysfs files to
follow the internal status is what the userspace should do.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 34bd1fdb4713..5d895154bcfa 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1602,7 +1602,7 @@ static int damon_sysfs_add_target(struct damon_sysfs_target *sys_target,
 			/* caller will destroy targets */
 			return -EINVAL;
 	}
-	/* todo: t->remove_on_commit = sys_target->obsolete */
+	t->remove_on_commit = sys_target->obsolete;
 	return damon_sysfs_set_regions(t, sys_target->regions, ctx->min_sz_region);
 }
 
@@ -1757,7 +1757,6 @@ static int damon_sysfs_commit_input(void *data)
 	if (err)
 		goto out;
 	err = damon_commit_ctx(kdamond->damon_ctx, param_ctx);
-	/* todo: re-populate targets directory without obsolete targets */
 out:
 	damon_destroy_ctx(test_ctx);
 	damon_destroy_ctx(param_ctx);
-- 
2.47.3

